packages/strapi-admin/admin/src/containers/Webhooks/EditView/index.js

```diff
@@ -0,0 +1,368 @@
+
+  const onCancelTrigger = () => {
+    abortController.abort();
+
+    dispatch({
+      type: 'ON_TRIGGER_CANCELED',
+    });
+  };
+
+  const handleReset = () =>
+    dispatch({
+      type: 'RESET',
+    });
+
+  const actions = [
+    {
+      color: 'primary',
+      disabled: triggerActionIsDisabled,
+      label: formatMessage({
+        id: `Settings.webhooks.trigger`,
+      }),
+      onClick: () => {
+        handleTrigger();
+      },
+      style: {
+        padding: '0 15px',
+      },
+      title: triggerActionIsDisabled
+        ? formatMessage({
+            id: `Settings.webhooks.trigger.save`,
+          })
+        : null,
+      type: 'button',
+      icon: (
+        <Play
+          width="6px"
+          height="7px"
+          fill={triggerActionIsDisabled ? '#b4b6ba' : '#ffffff'}
+        />
+      ),
+    },
+    {
+      color: 'cancel',
+      disabled: actionsAreDisabled,
+      label: formatMessage({
+        id: `app.components.Button.reset`,
+      }),
+      onClick: () => handleReset(),
+      style: {
+        padding: '0 20px',
+      },
+      type: 'button',
+    },
+    {
+      color: 'success',
+      disabled: actionsAreDisabled,
+      label: formatMessage({
+        id: `app.components.Button.save`,
+      }),
+      style: {
+        minWidth: 140,
+      },
+      type: 'submit',
+    },
+  ];
+
+  const headerProps = {
+    title: {
+      label: headerTitle,
+    },
+    actions: actions,
+  };
+
+  const handleBlur = () => {
+    if (submittedOnce) checkFormErrors();
+  };
+
+  const handleChange = ({ target: { name, value } }) => {
+    dispatch({
+      type: 'ON_CHANGE',
+      keys: name.split('.'),
+      value,
+    });
+  };
+
+  const handleClick = () => {
+    dispatch({
+      type: 'ADD_NEW_HEADER',
+      keys: ['headers'],
+    });
+  };
+
+  const handleRemove = ({ event, index }) => {
+    dispatch({
+      type: 'ON_HEADER_REMOVE',
+      index,
+      event,
+    });
+    resetError('headers');
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setSubmittedOnce(true);
+    checkFormErrors(true);
+  };
+
+  const submitForm = () => {
+    if (!isCreating) {
+      updateWebhook();
+    } else {
+      createWebhooks();
+    }
+  };
+
+  const checkFormErrors = async (submit = false) => {
+    const webhookToCheck = modifiedWebhook;
+    set(webhookToCheck, 'headers', cleanHeaders());
+
+    try {
+      await createYupSchema(form).validate(webhookToCheck, {
+        abortEarly: false,
+      });
+
+      setErrors({});
+      if (submit) submitForm();
+    } catch (err) {
+      setErrors(getYupInnerErrors(err));
+      if (submit) strapi.notification.error('notification.form.error.fields');
+    }
+  };
+
+  const cleanHeaders = () => {
+    const { headers } = modifiedWebhook;
+
+    if (Object.keys(headers).length === 1) {
+      const { key, value } = headers[0];
+      if (key.length === 0 && value.length === 0) return [];
+    }
+    return headers;
```

Comment:
My understanding is that headers is an array so I am not sure you reach the first condition either ways, I am not sure to understand why you need such logic anyway, can you detail a bit?

(If you could avoid inline ifs as well it would be great)

Reply:
Yes, `headers` is an array of objects to easily manage the DOM, the `add` and `remove` events and to display validations.

This logic exists because we can delete a header one by one with the `-` button, but if it's the first item, it just removes the values to keep the empty fields so a fake header exists.
But we don't want it to be sent to the backend as an object of empty values so it needs to be cleaned before submit.